// instagram-private-api/client/v1/feeds/account-followers.js
var _ = require('lodash');
var util = require('util');
var FeedBase = require('./feed-base');

function sleep_429(time) {
    var stop = new Date().getTime();
    while(new Date().getTime() < stop + time) {
        ;
    }
}

function AccountFollowersFeed(session, accountId, limit) {
    this.accountId = accountId;
    this.limit = limit || Infinity;
    this.timeout = 10 * 60 * 1000;
    FeedBase.apply(this, arguments);
}
util.inherits(AccountFollowersFeed, FeedBase);

module.exports = AccountFollowersFeed;
var Request = require('../request');
var Account = require('../account');

AccountFollowersFeed.prototype.get = function () {
    var that = this;
    return new Request(that.session)
        .setMethod('GET')
        .setResource('followersFeed', {
            id: that.accountId,
            maxId: that.cursor
        })
        .send()
        .then(function(data) {
            if(data.statusCode == 429){
                sleep_429(30000);
                return 429
            }

            sleep_429(1000)

            that.moreAvailable = !!data.next_max_id;
            if (that.moreAvailable) {
                that.setCursor(data.next_max_id);
            }

            return [_.map(data.users, function (user) {
                return new Account(that.session, user);
            }), data.next_max_id]
        })
};
